build.gradle

```
@@ -12,8 +12,8 @@ plugins {
 	id "com.diffplug.spotless" version "6.5.1"
 }
 
-sourceCompatibility = 1.11
-targetCompatibility = 1.11
+sourceCompatibility = 1.17
```

I'd suggest using the `JavaVersion.VERSION_17` constant for that.
```suggestion
sourceCompatibility = JavaVersion.VERSION_17
```